orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/tasks/EvaluateVariablesTask.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2018 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.orca.pipeline.tasks;
+
+import com.netflix.spinnaker.orca.ExecutionStatus;
+import com.netflix.spinnaker.orca.Task;
+import com.netflix.spinnaker.orca.TaskResult;
+import com.netflix.spinnaker.orca.pipeline.EvaluateVariablesStage;
+import com.netflix.spinnaker.orca.pipeline.model.Stage;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import javax.annotation.Nonnull;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@Component
+public class EvaluateVariablesTask implements Task {
+
+  @Autowired
+  public EvaluateVariablesTask() {
+  }
+
+  @Nonnull
+  @Override
+  public TaskResult execute(@Nonnull Stage stage) {
+    EvaluateVariablesStage.EvaluateVariablesStageContext context =
+      stage.mapTo(EvaluateVariablesStage.EvaluateVariablesStageContext.class);
+
+    Map<String, String> outputs = new HashMap<>();
+    for (EvaluateVariablesStage.Variable v : context.getVariables()) {
+      outputs.put(v.getKey(), v.getValue());
+    }
+
+    return new TaskResult(ExecutionStatus.SUCCEEDED, stage.mapTo(Map.class), outputs);
```

Comment:
you don't need the `stage.mapTo(Map.class)` here ... `Collections.emptyMap()` would suffice.